core/src/alpenglow_consensus/block_creation_loop.rs

```diff
@@ -110,6 +108,35 @@ enum StartLeaderError {
     VoteNotRooted,
 }
 
+fn start_receive_and_record_loop(
+    exit: Arc<AtomicBool>,
+    poh_recorder: Arc<RwLock<PohRecorder>>,
+    record_receiver: Receiver<Record>,
+) {
+    while !exit.load(Ordering::Relaxed) {
+        match record_receiver.recv_timeout(Duration::from_millis(100)) {
```

Comment:
might be nice to comment why we chose 100ms. I'm assuming this could just be a `recv` if not for the need to check exit flag at some frequency